### PR TITLE
Performance optimization

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Replace any() and all() builtins to plain for ... - it speedups performance in multiple invocations. Probably due context is not switched.

## How has this been tested?

- [x] performance test after #299

### CODE REVIEW

**Review by kmnls — APPROVED** [review]

File: credsweeper/app.py [hunk 1]
```diff
@@ -308,13 +310,18 @@ def file_scan(self, content_provider: Union[DiffContentProvider, TextContentProv
...
             ml_cred_groups = []
             for group_key, group_candidates in cred_groups.items():
                 # Analyze with ML if all candidates in group require ML
-                if all(candidate.use_ml for candidate in group_candidates):
+                has_to_be_added = True
+                for candidate in group_candidates:
+                    if not candidate.use_ml:
+                        has_to_be_added = False
+                        break
+                if has_to_be_added:
```
[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py:324]: there is ``all()`` optimization. Not ``any()``

[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/app.py]: below is double else...

**Review by kmnls — APPROVED** [review]

**Review by xDizzix — APPROVED** [review]

**Comment by babenek** [comment]
@csh519 
at least, 15% for the benchmark
3.10 Significantly speed-up. Was 203, now 168. Delta(%*10)=172
3.9 Significantly speed-up. Was 169, now 143. Delta(%*10)=153
3.8 Significantly speed-up. Was 171, now 145. Delta(%*10)=152

File: credsweeper/__main__.py [hunk 2]
```diff
@@ -7,8 +7,8 @@
 from typing import Any, Union, Optional, Dict
 
 from credsweeper import __version__
+from credsweeper.app import APP_PATH
 from credsweeper.app import CredSweeper
```
Comment (csh519) [inline credsweeper/__main__.py:11]:
```suggestion
from credsweeper.app import APP_PATH, CredSweeper
```

**Comment by csh519** [comment]
> @csh519 
> at least, 15% for the benchmark
> 3.10 Significantly speed-up. Was 203, now 168. Delta(%*10)=172
> 3.9 Significantly speed-up. Was 169, now 143. Delta(%*10)=153
> 3.8 Significantly speed-up. Was 171, now 145. Delta(%*10)=152

@babenek 
Maybe `static_keyword_checklist` seems to have helped improve performance!
Thank you for improving the performance.

**Review by csh519 — APPROVED** [review]
Thank you for optimization!

LGTM :+1:

**Review by kmnls — APPROVED** [review]

**Pull request merged**